src/meltano/core/state_store/s3.py

```diff
@@ -84,6 +84,9 @@ def client(self):
                 raise InvalidStateBackendConfigurationException(
                     "AWS access key ID configured, but no AWS secret access key."
                 )
+            else:
+                session = Session()
```

Comment:
`boto3.Session` is not imported at this point. Why is the import happening within the function?

Reply:
Ah, good catch. 

We want to lazily import boto3, since it's an optional dependency that should only be needed when `state_backend.uri` is set to a URI with `s3://` scheme. Importing within the method is obviously fragile (especially since this `else` branch is difficult to fully test for every potential method of providing credentials); is there a better pattern for this?